util/update_pcluster_configs.py

```diff
@@ -467,8 +405,8 @@ def _validate_documents_against_existing_version(args, files_to_upload, sts_cred
     for file in files_to_upload.keys():
         for region in args.regions:
             logging.info("Validating file %s in region %s", file, region)
-            doc_manager = S3JsonDocumentManager(region, sts_credentials.get(region))
-            current_file = doc_manager.download(args.bucket.format(region=region), FILE_TO_S3_PATH[file])
+            doc_manager = S3DocumentManager(region, sts_credentials.get(region))
+            current_file = json.loads(doc_manager.download(args.bucket.format(region=region), FILE_TO_S3_PATH[file]))
             logging.info("Current version: %s", current_file)
             logging.info("New version: %s", files_to_upload[file][region])
             doc_manager.validate_document(current_file, files_to_upload[file][region])
```

Comment:
This is not in DocumentManager anymore.

Can you please run a few tests:
- dryrun enabled
- without dryrun (and your bucket :))
- rollback tests

Reply:
optional: This kind of errors can be detected by flake8 or other static linters. it might be worth to enable flake8 in the util directory from tox config

Reply:
Thanks for the tip on `flake8` :D I've added the `../util/` directory and made a bunch of changes based on the results of that. Just FYI, flake8 did not catch this error.

Of course testing code fully does the trick.

Reply:
Here's how I've tested:

1. Dryrun / without: `python update_pcluster_configs.py --partition commercial --config-files feature_whitelist instances --efa-instance c5n.18xlarge --autodetect-regions --skip-validation --bucket [bucket] [--deploy]`
3. `python update_pcluster_configs.py --partition commercial --rollback --rollback-file-path rollback-data.json --bucket [bucket]`

I had to make a small change to `_generate_rollback_data()` to skip generating rollback data for objects that don't already exist.